core/tests/unit/ldml/test_transforms.cpp

```diff
@@ -1033,6 +1033,44 @@ test_normalize() {
   return EXIT_SUCCESS;
 }
 
+#include <unicode/uchar.h>
+
+#define NFD_FILE "common/web/types/src/ldml-keyboard/nfd-table.ts"
+int
+write_nfd_table() {
+  std::cout << " writing: " << NFD_FILE << std::endl;
+  FILE *f = fopen(NFD_FILE, "w");
+  assert(f != nullptr);
+  // we're going to need an NFD normalizer
+  UErrorCode status           = U_ZERO_ERROR;
+  const icu::Normalizer2 *nfd = icu::Normalizer2::getNFDInstance(status);
+  assert(U_SUCCESS(status));
+  fprintf(f, "// NFD hasBoundaryBefore\n");
+  fprintf(f, "// Generated by test_transforms.cpp\n");
+  fprintf(f, "// regenerate with: core/build/SOMETHING/SOMETHING/tests/unit/ldml/test_transforms --write-nfd\n");
+  fprintf(f, "export const ICU_VERSION='%s';\n", U_ICU_VERSION);
+  fprintf(f, "export const UNICODE_VERSION='%s';\n", U_UNICODE_VERSION);
+
+  fprintf(f, "export const nfdNoBoundaryBefore = [\n");
+  for (km_core_usv ch = 0; ch < 0x10FFFF; ch++) {
```

Comment:
Might as well start at 0x20 ... less weird than starting at 0, and looks more like a normal "Unicode Character Range".

```suggestion
  for (km_core_usv ch = 0x20; ch < 0x10FFFF; ch++) {
```

Reply:
why 0x20?  Nulls are characters too!

Reply:
ok, no worries. doesn't change output anyway.